tests/v2/actions/provisioning/creates.go

```diff
@@ -1071,3 +1071,28 @@ func DisableUpdateConfig(client *rancher.Client) {
 		return nil
 	})
 }
+
+// CreateProvisioningRKE1ClusterWithClusterTemplate provisions an rke1 cluster by using the rke1 template and revision ID and other values from the template.
+func CreateProvisioningRKE1ClusterWithClusterTemplate(client *rancher.Client, provider RKE1Provider, rke1Cluster *management.Cluster, clusterConfig *clusters.ClusterConfig, nodeTemplate *nodetemplates.NodeTemplate) (*management.Cluster, error) {
```

Comment:
Can this be made to match the behavior of CreateProvisioningRKECluster more closely. notably:
* create the rke1Cluster management object in this function, not sending the premade object. Can you send the ClusterTemplate object instead?
* why is nodeTemplate required? I see that it i used in nodepool setup but, I was assuming that was done as part of the clusterTemplate?

Reply:
This function is focused on creating the cluster itself, and part of the requirement is to include specific configurations that are added as part of the cluster config. The ClusterTemplateID and RevisionID will be integrated into this configuration when the cluster is being created.

Regarding the node pools, while the cluster template provides the configuration, node pools are not created at the same time as the template. Node pools are only instantiated when the cluster itself is created, and the cluster then references the relevant cluster template. This is why we need the nodeTemplate object—it is essential for setting up node pools during cluster creation, not during the template definition.

Reply:
node pools - acknowledged and makes sense. 

I understand the purpose of this function, but we should still send it relevant and concise data, and be consistent with other functions. We don't send the management.Cluster object in the normal createRKE1cluster func, so we shouldn't do it here either. So this func should take in the object specified in the name, clusterTemplate, then create management.Cluster, then ask rancher to create that object.

Reply:
I'm not sure I understand. we do send in the management.Cluster object in ProvisioningRKE1Cluster. https://github.com/rancher/rancher/blob/main/tests/v2/actions/provisioning/creates.go#L398. also we are using the CreateRKE1Cluster function to create the cluster. this is more or less the same on what CreateProvisioningRKE1Cluster is doing except in this we are sending in the cluster template ID and revision ID as well.

Reply:
in that function, rkeCluster is not a parameter to the current rke1Create call, rather it is built as part of the function and used in the shepherd call.

Reply:
if I understand it correctly, we want to send in the template parameters and in the creates.go, we build the cluster object in the same Create function and then provision the cluster. correct?

Reply:
correct